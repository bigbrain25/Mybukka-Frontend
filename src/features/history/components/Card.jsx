import React from 'react';
import PropTypes from 'prop-types';
import Container from 'Components/container';
import Body from '../card/Body';
import Head from '../card/Head';
import Footer from '../card/Footer';

// eslint-disable-next-line max-len
const Col = ({ children, classNames }) => <div className={`col-md-11 mx-auto ${classNames}`}>{children}</div>;

const Card = ({
  handleClick,
  time,
  orderId,
  mealTitle,
  price,
  status,
  quantity
}) => (
  <div
    className="d-block d-md-block d-xl-none d-lg-none bg-white py-3 mb-3
    cursor-pointer"
    onClick={handleClick}
    role="button"
    data-testid="card-button"
    tabIndex="0"
    aria-pressed
  >
    <Container
      classNames="d-flex-column column mb-2 justify-content-center"
    >
      <Col>
        <Head orderId={orderId} time={time} />
      </Col>
      <hr className="border-line border-box col-12" />
      <Col classNames="pt-1 pb-1">
        <Body quantity={quantity} title={mealTitle} />
      </Col>
      <Col>
        <Footer status={status} price={price} />
      </Col>
    </Container>
  </div>
);

export default Card;

Card.propTypes = {
  handleClick: PropTypes.func.isRequired,
  time: PropTypes.string.isRequired,
  orderId: PropTypes.string.isRequired,
  mealTitle: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  quantity: PropTypes.string.isRequired,
};

Col.defaultProps = {
  classNames: ''
};

Col.propTypes = {
  classNames: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};
