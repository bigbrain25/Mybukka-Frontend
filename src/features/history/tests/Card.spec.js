import React from 'react';
import Card from '../components/Card';

describe('Card component', () => {
  const props = {
    quantity: '2',
    mealTitle: 'salad and bbq',
    handleClick: jest.fn(),
    time: '27/3/2019 12:43 pm',
    orderId: '03d77',
    price: '3000',
    status: 'pending',
  };

  const { container, getByRole } = render(<Card {...props} />);

  it('should renders properly', () => {
    expect(container).toMatchSnapshot();
  });

  it('should triggers the handleclick function when clicked', () => {
    const button = getByRole('button');
    fireEvent.click(button);

    expect(props.handleClick).toBeCalled();
  });
});
