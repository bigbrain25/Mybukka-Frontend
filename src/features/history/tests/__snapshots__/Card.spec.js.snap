// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should renders properly 1`] = `
<div>
  <div
    aria-pressed="true"
    class="d-block d-md-block d-xl-none d-lg-none bg-white py-3 mb-3
 cursor-pointer"
    role="button"
    tabindex="0"
  >
    <div
      class="container px-0 d-flex-column column mb-2 justify-content-center"
    >
      <div
        class="col-md-11 mx-auto "
      >
        <div
          class="row justify-content-between text-uppercase"
        >
          <h5
            class="font-size-14"
          >
            #
            03d77
          </h5>
          <h5
            class="font-size-14 suggested-item-price"
          >
            27/3/2019 12:43 pm
          </h5>
        </div>
      </div>
      <hr
        class="border-line border-box col-12"
      />
      <div
        class="col-md-11 mx-auto pt-1 pb-1"
      >
        <div
          class="row justify-content-between"
        >
          <h5
            class="text-uppercase font-size-14"
          >
            2
            X  
            salad and bbq
             ...
          </h5>
        </div>
      </div>
      <div
        class="col-md-11 mx-auto "
      >
        <div
          class="row justify-content-between text-capitalize"
        >
          <h5
            class="font-size-14"
          >
            Status: 
            <span
              class="text-warning"
            >
               
              pending
            </span>
          </h5>
          <h5
            class="text-color font-size-14"
          >
             â‚¦
            3000
            .00
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
